Automattic/wp-calypso
# FSE plugin: add newspack posts block to version control

Background pbAok1-dw-p2

#### Changes proposed in this Pull Request

* Adds Newspack plugin to FSE `package.json`
* Builds the Newspack posts block via `prepare` script
* Newspack plugin has its own `package.json` with `dependencies` array — which is fantastic because now we get dependencies pulled into our package-lock. I don't think we were listing those dependencies anywhere even if we had the code checked in, which seems fragile.

##### TODO
- [x] Do something about the asset version — maybe we can just use the FSE version here? Alternatively, pull Newspack plugin version from package.json. _We are updating the version from the package.json._
- [x] Someone could be pinged when new plugin release is available by adding `reviewers` or `assignees` to `renovate.json` for this dependency.

#### Testing instructions

- Run `npm distclean`, confirm that `full-site-editing-plugin/blog-posts-block/newspack-homepage-articles` is gone.
- Run `npm i` (at the Calypso root) and confirm that you get freshly built block in the above directory
- Confirm that the block works and is renamed appropriately
- Confirm that updating the version of Newspack in the package.json and then running `npm i` also updates the version of Newspack locally -- includes changing the version defined in full-site-editing-plugin/blog-posts-block/index.php.
